packages/termux-tools/pkg

```diff
@@ -54,25 +59,44 @@ select_mirror() {
 		return
 	fi
 
-	if [ -n "$(find /data/data/com.termux/cache/apt/pkgcache.bin -mmin -180 2>/dev/null)" ]; then
-		# Mirrors are rotated if either they are not working or 180 minutes timeout
-		# has passed.
+	# Mirrors are rotated if 24 hours timeout has been passed or mirror is no longer accessible.
+	if [ -n "$(find /data/data/com.termux/cache/apt/pkgcache.bin -mmin -1440 2>/dev/null)" ]; then
 		if [ -n "${current_mirror}" ] && curl --user-agent 'Termux-PKG/1.0 mirror-checker' --head --fail --location "${current_mirror}/dists/stable/Release" >/dev/null 2>&1; then
 			echo "Reusing mirror: ${current_mirror}"
 			return
 		fi
 	fi
 
-	# Shuffle mirrors and pick the first one which is accessible.
-	local m selected_mirror=""
-	for m in $(echo "$top_mirrors" | tr ' ' '\n' | shuf); do
-		echo "Trying mirror: $m"
+	# Test mirror availability, remove unaccessible mirrors from list.
+	echo "Testing the available mirrors (once in 24h):"
+	local m total_mirror_weight=0
+	for m in "${!mirrors[@]}"; do
+		echo -n "[*] ${m}: "
 		if curl --user-agent 'Termux-PKG/1.0 mirror-checker' --head --fail --location "${m}/dists/stable/Release" >/dev/null 2>&1; then
-			selected_mirror="$m"
-			break
+			echo "ok"
+			total_mirror_weight=$((total_mirror_weight + mirrors[$m]))
+		else
+			echo "broken"
+			unset "mirror[$m]"
 		fi
 	done
 
+	# Weight-based mirror selection.
+	local selected_mirror=""
+	if ((total_mirror_weight > 0)); then
+		local random_weight calc_weight=0
+		random_weight=$((RANDOM % total_mirror_weight + 1))
+		for m in "${!mirrors[@]}"; do
```

Comment:
Here is an issue that $mirrors array is not sorted by value which will produce wrong results. I will fix that later and merge the update.